sklearn/neural_network/_multilayer_perceptron.py

```diff
@@ -327,36 +343,46 @@ def _backprop(self, X, y, activations, deltas, coef_grads, intercept_grads):
         loss_func_name = self.loss
         if loss_func_name == "log_loss" and self.out_activation_ == "logistic":
             loss_func_name = "binary_log_loss"
-        loss = LOSS_FUNCTIONS[loss_func_name](y, activations[-1])
+        loss = LOSS_FUNCTIONS[loss_func_name](y, activations[-1], sample_weight)
         # Add L2 regularization term to loss
         values = 0
         for s in self.coefs_:
             s = s.ravel()
             values += np.dot(s, s)
-        loss += (0.5 * self.alpha) * values / n_samples
+        if sample_weight is None:
+            n_sum = n_samples
+        else:
+            n_sum = sample_weight.sum()
```

Comment:
Let's rename this variable and the corresponding argument in `_compute_loss_grad` to `sw_sum` to be more descriptive and consistent with the naming convention used in `_linear_loss.py`.